transip/client.go

```diff
@@ -0,0 +1,103 @@
+package transip
+
+import (
+	"context"
+	"time"
+
+	"github.com/transip/gotransip"
+	transipdomain "github.com/transip/gotransip/domain"
+	"github.com/libdns/libdns"
+)
+
+func (p *Provider) setupRepository() error {
+	if p.repository == nil {
```

Comment:
I think this is a good idea, but it just occurred to me that this isn't safe for concurrent use. (sync.Mutex would be helpful here.) The wiki doesn't mention this right now because I didn't think of it before, but I bet it'd be a good idea for each method to be safe for concurrent use.

That's my last bit of feedback. Otherwise, LGTM!

Reply:
I updated the wiki and added the recommendations for thread safety, including the use of [sync.Once](https://golang.org/pkg/sync/#Once) if that makes things easier.